### ISSUE TITLE
Non ONA managed CNAMES (external DNS references)

### ISSUE THREAD

**Issue by rdorman** [opennetadmin/ona#70]
I need to be able to create a CNAME from one of my domains to an Internet domain (in this case an AWS EC2 Elastic IP).  I cannot do that because it has to be an "existing A record" that is within the ONA database.  Adding amazonaws.com or something of that nature just wont scale.

**Comment by mattpascoe** [comment]
Capturing another lame workaround thought. (yes one of these days I'll actually fix this problem)

Add a host called 'header-file.example.com' (use whatever domain you have in your system).. Add an IP address that makes sense, something out of the way and un-used.  I used 192.168.0.10.

Then you could add a CNAME like 'mycname.example.com' that points to header-file.example.com.  MAKE sure that both of these DNS records are marked as disabled.  I'd also suggest putting good notes for these hosts and dns records that indicate they are actually defined in the header file.

Basically this is a cheesy way to get records in ONA that can be searched but that are disabled and do not create DNS configurations. It helps to point you at the fact that there are magical hidden records in a file outside of ONA data. You still of course need to add the REAL records into your header file. 

This is minimal stop gap until I can add the header file blob to the zone and or the override record idea.

**Comment by dmgeurts** [comment]
This is an old thread, dare I ask if there's any news on this?

Background: Just ran into this issue. I have ONA for home networking and FreeIPA for (home) business stuff. I have a firewall that crosses both domains and thus was hoping to add a CNAME into ONA (with a short local name for users (lazy me)) to an A record that resides in the DNS server on the FreeIPA server. ONA won't let me, great to have such a consistency check, but I'd love to either disable the sanity check on my ONA server or be allowed to override and accept the risk for this one CNAME record.

Pretty please?

**Comment by mattpascoe** [comment]
For sure this is a long standing issue and I've not resolved it as I had hoped to.  It comes down to time and effort and this one is a bugger to do 'right'.  I had started down a path that I think would be viable.  It's probably in the 50% complete range.  heck the name of the branch indicates that I saw it as finally addressing an old problem. (https://github.com/opennetadmin/ona/tree/feature_finaly).

I won't try and mislead you.. It's a busy time for me right now and I can't even keep up with some of my higher priority things in life.  My desire is certainly to work on this and get a solution as it is probably the #1 feature to implement.  These reminders keep it fresh in my mind so I say keep hounding/reminding me! ;)

This is the "workaround" using footers but it is of course not optimal either: https://github.com/opennetadmin/build_bind#fetching-zone-footers-from-remote-web-server

Hopefully that could be an option.  I would also suggest possibly using the tinydns backend (https://github.com/opennetadmin/build_tinydns) as it has a much simpler footer type solution.  This is actually what I use at $DAYJOB and probably why I've not been forced into fixing this mess.

Anyway, I hope this non answer might give you viable workarounds.  And you are reminding me that I STILL need to address this issue.

**Comment by dmgeurts** [comment]
That's the issue with todo lists, they seem to only get longer and never shorter...! I feel your pain.

As it's a single entry, I created an A record that isn't served by ONA, but happily accepted into config, just so I can point the cname there. Dirty, but it works...
